Source/Scene/ModelExperimental/ModelExperimentalAnimationCollection.js

```diff
@@ -0,0 +1,535 @@
+ * @internalConstructor
+ * @class
+ *
+ * @see ModelExperimental#activeAnimations
+ */
+function ModelExperimentalAnimationCollection(model) {
+  /**
+   * The event fired when an animation is added to the collection.  This can be used, for
+   * example, to keep a UI in sync.
+   *
+   * @type {Event}
+   * @default new Event()
+   *
+   * @example
+   * model.activeAnimations.animationAdded.addEventListener(function(model, animation) {
+   *   console.log('Animation added: ' + animation.name);
+   * });
+   */
+  this.animationAdded = new Event();
+
+  /**
+   * The event fired when an animation is removed from the collection.  This can be used, for
+   * example, to keep a UI in sync.
+   *
+   * @type {Event}
+   * @default new Event()
+   *
+   * @example
+   * model.activeAnimations.animationRemoved.addEventListener(function(model, animation) {
+   *   console.log('Animation removed: ' + animation.name);
+   * });
+   */
+  this.animationRemoved = new Event();
+
+  this._model = model;
+  this._runtimeAnimations = [];
+  this._previousTime = undefined;
+}
+
+Object.defineProperties(ModelExperimentalAnimationCollection.prototype, {
+  /**
+   * The number of animations in the collection.
+   *
+   * @memberof ModelExperimentalAnimationCollection.prototype
+   *
+   * @type {Number}
+   * @readonly
+   */
+  length: {
+    get: function () {
+      return this._runtimeAnimations.length;
+    },
+  },
+
+  /**
+   * The model that owns this animation collection.
+   *
+   * @memberof ModelExperimentalAnimationCollection.prototype
+   *
+   * @type {ModelExperimental}
+   * @readonly
+   */
+  model: {
+    get: function () {
+      return this._model;
+    },
+  },
+});
+
+function addAnimation(collection, animation, options) {
+  const model = collection._model;
+  const runtimeAnimation = new ModelExperimentalAnimation(
+    model,
+    animation,
+    options
+  );
+  collection._runtimeAnimations.push(runtimeAnimation);
+  collection.animationAdded.raiseEvent(model, runtimeAnimation);
+  return runtimeAnimation;
+}
+
+/**
+ * Creates and adds an animation with the specified initial properties to the collection.
+ * <p>
+ * This raises the {@link ModelExperimentalAnimationCollection#animationAdded} event so, for example, a UI can stay in sync.
+ * </p>
+ *
+ * @param {Object} options Object with the following properties:
+ * @param {String} [options.name] The glTF animation name that identifies the animation. Must be defined if <code>options.index</code> is <code>undefined</code>.
+ * @param {Number} [options.index] The glTF animation index that identifies the animation. Must be defined if <code>options.name</code> is <code>undefined</code>.
+ * @param {JulianDate} [options.startTime] The scene time to start playing the animation.  When this is <code>undefined</code>, the animation starts at the next frame.
+ * @param {Number} [options.delay=0.0] The delay, in seconds, from <code>startTime</code> to start playing.
```

Comment:
This is another parameter that seems to be overridden by looping the animation: [Sandcastle](http://localhost:8080/Apps/Sandcastle/index.html#c=xVf/b+I2FP9XLH6CiToJUGhZr1pFe71Ora4r6CZtmXomMcRqYke2Q+lN/d/3bCchQG9aVfWGEPjLsz+fz3sv9kskuNJoxegjlegD4vQRTahiRYa/2LF22IpsfyK4JoxTGba66O+QI6QSUaTxGWcZ0XSMtCxoN+TPnZ9DHnK3I45SET3gqJCScj1jGQWMcv9fi5QRfg5r8UKK7EqJo6EftM3OYavn94KDIDjwj2c9f9wbjfsj7AdDfzgYjA77g6Ph0eC4P/ojbIXc4UVWR1QoLbJpQuJdNZPGTNvSX0iyzICVG5vRtR6jr2ZiJVhcz96A5vbHsnPF80KjhbL/XcS4gG70LbvPREzTG5AiGUlRVjY6zk+oHsAxWywKZZywolG/Xe6EidaSzQtNFdZ0PRFCxvd+F/nYN+IQejY/X2vfeh66MYCqkm3hFez6pzEsQbVVFLbKAMVoBhT4MqUQQGcBa2lKI7BaFDzSTHDUrjlDfO2kRWpXYyY23jSnkfIgcsRzPLzLVC+uhfGjV8FVaN4ynX3cG8VLWFETMZ/Dqukkg2g76X53JE1ZlqcUnCDz5H3VOCQL5IQ0Bt6moQ6L3QtNivkPCozFM3DbkamHXynrv0lD7d8KwjX7RuPO/yHzoIZ/s+CXc3H6wPiPSEWD08xE03+bAHdAohvyZv4Ye/CdEsPLinDnkucQAKBs3UMTOM+3KPuvoHwuSSTQRGS5pErZRAOg96Jv0TZglRyWPsxkAbfbnvdfJ0Vw+r6OtxDN9p7vg9fky6UUBY/RF5qwKH036g6lBNnuvSl1rjhcxblIiWU4o0q/w1O7BWIw9kf2RPxLCP5qVDhJvlPYfKIkZnx5y3SU3InUcSvnoCJJsBZ3YEG4agdHfsdu6rtfVzu5fRdsTWOoczI6k2C7EDLblGr1kMJQz5HU2YmP9ZpLCoUh0UKW5RsXUlc3c9h6tF6uKrXauU2HFjLtooSyZaJLlztauVDMGtdcJkRCnQR6+rZsPKdLSakqI3IQ9PrYHw0Gw+C49O1ggP1Dvz/yh+WAQzFtxweV1S9WEajAuWQZQK6gFpM0Eyt6Bj51MWkUW6aC+84qEsclmyoKZsHFOocS0BSRUAYa4uZkb9fJZc6QMTJeqEaqklKy9fiFOCQ7YW9Go5Gelf8aiQYZ1OiVO1+kKcsVFL7498vp0aBh8EJi1JlamzXr7vFWb6OHwb1bZLewXTqFi3iMhhuYjKzN5BRyCyZ6PnzKuedOM1LWKRAXEj/dgg+ZolgnlLc3D6y1qJ/a6i0E2EsCuSseznTtnT3ffO+pgoq16VN/3307D9pBcNhpGDnWc3OEwZ7TPKGSYgm2hUI/NR5816i0VutIZPLKVS3A1iaYyck6dzxP0hWVavMO5sZhMXkaowBvCBtTk9Of+VSLfMce3JOPt5K2Br2GKXx3cXtxNqvj4t5MzN+zITwlPI6I0nA+A8GZEOmcyBvKCxcS5WS1uq0TpZ9Selqh/gJlDJwWJvfbcBNoCjcBlGfKm8PdSjWOlKpOxBOvufQkZivE4g8vvKCiKCVKwcyiSG2yha3TEw/s95amwsb5M/gPnGXMkuD02g1ijE886L68UjuFOzv/Aw)

again, I would document this since it might not be obvious (I was expecting it to delay for a bit then start the loop)